ocs_ci/deployment/deployment.py

```diff
@@ -237,38 +243,46 @@ def get_olm_and_subscription_manifest():
         templating.dump_data_to_temp_yaml(
             subscription_yaml_data, subscription_manifest.name
         )
-        return olm_manifest.name, subscription_manifest.name
-
-    def deploy_ocs_via_operator(self):
-        """
-        Method for deploy OCS via OCS operator
-        """
-        logger.info("Deployment of OCS via OCS operator")
-        olm_manifest, subscription_manifest = (
-            self.get_olm_and_subscription_manifest()
-        )
-        self.label_and_taint_nodes()
-        run_cmd(f"oc create -f {olm_manifest}", timeout=2400)
-        catalog_source = CatalogSource(
-            resource_name='ocs-catalogsource',
-            namespace='openshift-marketplace',
-        )
-        # Wait for catalog source is ready
-        catalog_source.wait_for_state("READY")
-        run_cmd(f"oc create -f {subscription_manifest}")
+        run_cmd(f"oc create -f {subscription_manifest.name}")
         # wait for package manifest
         package_manifest = PackageManifest(
             resource_name=defaults.OCS_OPERATOR_NAME
         )
         # Wait for package manifest is ready
         package_manifest.wait_for_resource(timeout=300)
         channel = config.DEPLOYMENT.get('ocs_csv_channel')
-        csv_name = package_manifest.get_current_csv(channel=channel)
         subscription_plan_approval = config.DEPLOYMENT.get(
             'subscription_plan_approval'
         )
         if subscription_plan_approval == 'Manual':
             wait_for_install_plan_and_approve(self.namespace)
+
+    def deploy_ocs_via_operator(self):
+        """
+        Method for deploy OCS via OCS operator
+        """
+        ui_deployment = config.DEPLOYMENT.get('ui_deployment')
+        if ui_deployment:
+            logger.info("Preparing stuff for deployment by openshift-console")
+        else:
+            logger.info("Deployment of OCS via OCS operator")
+            self.label_and_taint_nodes()
+        run_cmd(f"oc create -f {constants.OLM_YAML}")
+        self.create_catalog_source()
+        if ui_deployment:
+            logger.info(
```

Comment:
This code can move into after line 265: as they are same if statement.

Reply:
Cannot cause I need to do all the operation above

Reply:
There are two if statements and one can go away?

Reply:
Look at the code @vasukulkarni , I need to print some message at the start, in case of UI deployment I want to inform different way that preparing stuff for deployment via openshift-console, in case of non UI deployment I need to also call function for labeling nodes, than I need to run command on line 270 and 271 in both UI and non UI deployment and than here in case of UI deployment I need to exit (return here).
So here need to be the condition once more I cannot move the return or info message above cause I still need to do what is done on lines 270 and 271.   So what is the issue here?

Reply:
I see it merged, but this is what I am suggesting for better reaadability (the first *if* as I read did nothing but print some info and was handled in second *if*) https://github.com/red-hat-storage/ocs-ci/pull/1129